packages/ckeditor5-code-block/src/utils.ts

```diff
@@ -258,3 +259,32 @@ export function canBeCodeBlock( schema: Schema, element: Element ): boolean {
 
 	return schema.checkChild( element.parent as Element, 'codeBlock' );
 }
+
+/**
+ * Get the translated message read by the screen reader when you enter or exit an element with your cursor.
+ */
+export function getCodeBlockAriaAnnouncement(
+	t: LocaleTranslate,
+	languageDefs: Array<CodeBlockLanguageDefinition>,
+	element: Element,
+	direction: 'enter' | 'leave'
+): string {
+	const languagesToLabels = getPropertyAssociation( languageDefs, 'language', 'label' );
+	const codeBlockLanguage = element.getAttribute( 'language' ) as string;
+
+	if ( codeBlockLanguage in languagesToLabels ) {
+		const language = languagesToLabels[ codeBlockLanguage ];
+
+		if ( direction === 'enter' ) {
+			return t( 'Entering %0 code snippet', language );
+		}
+
+		return t( 'Leaving %0 code snippet', language );
+	}
+
+	if ( direction === 'enter' ) {
+		return t( 'Entering code snippet' );
+	}
+
+	return t( 'Leaving code snippet' );
```

Comment:
I think we can safely assume there will never be a code snippet without the language upcasted in the editor. See https://github.com/ckeditor/ckeditor5/blob/84d1ada4e42bbdd3df3f2b4623a77a77db5d863f/packages/ckeditor5-code-block/src/converters.ts#L205-L207. I'd assume that `element.getAttribute( 'language' )!`.

Reply:
It can happen if `language` passed to `codeBlock` has no language definition specified in config (like `FooBar`). I'd stay with a bit more verbose but safer approach.